clippy_lints/src/mut_reference.rs

```diff
@@ -49,9 +49,8 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnnecessaryMutPassed {
                 }
             },
             ExprMethodCall(ref name, _, ref arguments) => {
-                let method_call = MethodCall::expr(e.id);
-                let method_type = borrowed_table.method_map.get(&method_call).expect("This should never happen.");
-                check_arguments(cx, arguments, method_type.ty, &name.node.as_str())
+                let method_type = borrowed_table.expr_ty(e);
+                check_arguments(cx, arguments, method_type, &name.node.as_str())
```

Comment:
`expr_ty(e)` gives you the returned type not the method type (which you have to get from its `DefId` now).